README.md

```diff
@@ -104,34 +104,84 @@ if(verifier.canQuickVerify(bag)){
 
 ##### Add other checksum algorithms
 
-You only need to implement 2 interfaces:
+You only need to implement 1 interface:
 
 ```java
-public class MyNewSupportedAlgorithm implements SupportedAlgorithm {
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.util.Formatter;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gov.loc.repository.bagit.exceptions.UnsupportedAlgorithmException;
+
+public class MyNewHasher implements Hasher {
+  static {
+    if (Security.getProvider("BC") == null) {
+      Security.addProvider(new BouncyCastleProvider());
+    }
+  }
+  
+  private static final Logger logger = LoggerFactory.getLogger(SHA1Hasher.class);
+  private MessageDigest messageDigest;
+  private final String bagitName;
+  private final String messageDigestName;
+  
+  public MyNewHasher(final String messageDigestName, final String bagitName) throws NoSuchAlgorithmException {
+    this.messageDigestName = messageDigestName;
+    this.bagitName = bagitName;
+    messageDigest = MessageDigest.getInstance(messageDigestName);
+  }
+
   @Override
-  public String getMessageDigestName() {
-    return "SHA3-256";
+  public void update(byte[] buffer, int length) {
+    messageDigest.update(buffer, 0, length);
   }
+
   @Override
-  public String getBagitName() {
-    return "sha3256";
+  public void clear() {
+    try {
+      messageDigest = MessageDigest.getInstance(messageDigestName);
+    } catch (NoSuchAlgorithmException e) {
+      logger.error("Could not get a new instance of the {} message digest", messageDigestName, e);
+    }
   }
-}
 
-public class MyNewNameMapping implements BagitAlgorithmNameToSupportedAlgorithmMapping {
   @Override
-  public SupportedAlgorithm getMessageDigestName(String bagitAlgorithmName) {
-    if("sha3256".equals(bagitAlgorithmName)){
-      return new MyNewSupportedAlgorithm();
+  public String getCalculatedValue() {
+    final Formatter formatter = new Formatter();
+    
+    for (final byte b : messageDigest.digest()) {
+      formatter.format("%02x", b);
     }
+    
+    final String hash = formatter.toString();
+    formatter.close();
+    
+    return hash;
+  }
 
-    return StandardSupportedAlgorithms.valueOf(bagitAlgorithmName.toUpperCase());
+  @Override
+  public String getBagitName() {
+    return bagitName;
+  }
+
+  @Override
+  public Hasher instanceOf() throws UnsupportedAlgorithmException {
+    try {
+      return new MyNewHasher(messageDigestName, bagitName);
+    } catch (NoSuchAlgorithmException e) {
+      throw new UnsupportedAlgorithmException(e);
+    }
   }
 }
 ```
 
-and then add the implemented `BagitAlgorithmNameToSupportedAlgorithmMapping`
-class to your `BagReader` or `bagVerifier` object before using their methods.
+and then add the implemented `Hasher` to the `bagitNameToHasherMap` in `BagCreator`, `BagVerifier` or 
```

Comment:
I feel like this block of text should be moved above the code block so it doesn't require you to connect text before and after almost 100 lines of code.